tests/cloner_test.go

```diff
@@ -1135,6 +1135,90 @@ var _ = Describe("all clone tests", func() {
 		})
 
 	})
+
+	var _ = Describe("Preallocation", func() {
+		f := framework.NewFramework(namespacePrefix)
+
+		var sourcePvc *v1.PersistentVolumeClaim
+		var targetPvc *v1.PersistentVolumeClaim
+		var cdiCr cdiv1.CDI
+		var cdiCrSpec *cdiv1.CDISpec
+
+		BeforeEach(func() {
+			By("[BeforeEach] Saving CDI CR spec")
+			crList, err := f.CdiClient.CdiV1beta1().CDIs().List(context.TODO(), metav1.ListOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			Expect(len(crList.Items)).To(Equal(1))
+
+			cdiCrSpec = crList.Items[0].Spec.DeepCopy()
+			cdiCr = crList.Items[0]
+
+			By("[BeforeEach] Forcing Host Assisted cloning")
+			var cloneStrategy cdiv1.CDICloneStrategy = cdiv1.CloneStrategyHostAssisted
+			cdiCr.Spec.CloneStrategyOverride = &cloneStrategy
+			_, err = f.CdiClient.CdiV1beta1().CDIs().Update(context.TODO(), &cdiCr, metav1.UpdateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		AfterEach(func() {
+			if sourcePvc != nil {
+				By("[AfterEach] Clean up source PVC")
+				err := f.DeletePVC(sourcePvc)
+				Expect(err).ToNot(HaveOccurred())
+			}
+			if targetPvc != nil {
+				By("[AfterEach] Clean up target PVC")
+				err := f.DeletePVC(targetPvc)
+				Expect(err).ToNot(HaveOccurred())
+			}
+
+			By("[AfterEach] Restoring CDI CR spec to original state")
+			crList, err := f.CdiClient.CdiV1beta1().CDIs().List(context.TODO(), metav1.ListOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			Expect(len(crList.Items)).To(Equal(1))
+
+			newCdiCr := crList.Items[0]
+			newCdiCr.Spec = *cdiCrSpec
+			_, err = f.CdiClient.CdiV1beta1().CDIs().Update(context.TODO(), &newCdiCr, metav1.UpdateOptions{})
+			Expect(err).ToNot(HaveOccurred())
```

Comment:
Same here but reverse, make sure the original value was restored.

Reply:
Done